Scene.pickPosition in 2D/CV

bagnell opened the pull request:
Adds pick position support in 2D and Columbus view. Fixes #4084.

- [x] Tests
- [x] Update CHANGES.md

pjcozzi commented on Source/Scene/Scene.js:
```diff
@@ -2760,17 +2748,60 @@ define([
+     * @returns {Cartesian3} The cartesian position.
+     *
+     * @exception {DeveloperError} Picking from the depth buffer is not supported. Check pickPositionSupported.
+     * @exception {DeveloperError} 2D is not supported. An orthographic projection matrix is not invertible.
```
This exception is not longer valid, right?

pjcozzi commented on Source/Scene/Scene.js:
```diff
@@ -2699,17 +2699,7 @@ define([
-     * @exception {DeveloperError} 2D is not supported. An orthographic projection matrix is not invertible.
-     */
-    Scene.prototype.pickPosition = function(windowPosition, result) {
+    Scene.prototype._pickPosition = function(windowPosition, result) {
```
Perhaps it would be more clear to name this `pickPosition3D` and make it `@private`.

pjcozzi replied:
Or whatever a more precise name is.

pjcozzi commented:
Can we also document that the results can be different in 2D vs. 3D/CV due to the orthographic projection and give some sense as to how the precision works?

For example, in 3D:

![image](https://cloud.githubusercontent.com/assets/782098/22845821/78cf70ee-efb2-11e6-886f-c2f62ed4389f.png)

In 2D:

![image](https://cloud.githubusercontent.com/assets/782098/22845853/99870608-efb2-11e6-8cdb-6a757704b174.png)

bagnell commented:
@pjcozzi This is ready for another look.

pjcozzi commented:
Do you get this test failure?
```
Scene/Scene pickPosition
Expected Cartesian3({ x: -455845.5482883507, y: -5210338.458916779, z: 3637550.496413719 }) to equal epsilon Cartesian3({ x: -455845.46867895435, y: -5210337.548977215, z: 3637549.8562320103 }), 1e-7.
Error: Expected Cartesian3({ x: -455845.5482883507, y: -5210338.458916779, z: 3637550.496413719 }) to equal epsilon Cartesian3({ x: -455845.46867895435, y: -5210337.548977215, z: 3637549.8562320103 }), 1e-7.
    at stack (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1475:17)
    at buildExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1445:14)
    at Spec.expectationResultFactory (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:584:18)
    at Spec.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:332:34)
    at Expectation.addExpectationResult (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:528:21)
    at Expectation.toEqualEpsilon (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1399:12)
    at http://localhost:8080/Specs/Scene/SceneSpec.js:660:30
    at compare (http://localhost:8080/Specs/addDefaultMatchers.js:255:29)
    at Expectation.toRenderAndCall (http://localhost:8080/ThirdParty/jasmine-2.2.0/jasmine.js:1378:35)
    at http://localhost:8080/Specs/Scene/SceneSpec.js:658:23
```

bagnell commented:
>Do you get this test failure?

I was not seeing any failures. I changed the test to check the values returned instead of that it was defined. I updated the epsilon values so they should pass now.

pjcozzi commented:
Very nice.
